### ISSUE TITLE
objection: join forces?

### ISSUE THREAD

**Issue by blshkv** [shroudedcode/apk-mitm#16]
I'm not sure if you are aware.
There is a tool https://github.com/sensepost/objection. It does similar patching and support both apk and ipa. It's written on a different language (python) but has more functions (frida integration etc).

So I was wondering, why don't you push your hard work to them and create a single place for all mitm tricks?

**Comment by shroudedcode** [comment]
I was not aware, but it does look very interesting and I'll certainly give it a try.

Here's why I think both projects deserve to coexist regardless:

- `objection` is quite a bit more difficult to get started with.

  While the README states that the installation is "simply a matter of `pip3 install objection`", actually using `objection patchapk` requires [installing four separate tools](https://github.com/sensepost/objection/wiki/Patching-Android-Applications#patching---dependencies) beforehand (in addition to Python, Java, and `objection` itself). `apk-mitm`, on the other hand, only requires Java and Node.js to be installed, all other dependencies are bundled with the package.

- `objection` can do a lot more which is both a good and a bad thing.

  I've used `apk-mitm` to reverse-engineer the APIs behind a lot of mobile apps and it has served me quite well. I have to admit that I've never even used Frida and that's not because I think it's bad but just because I never had the need for it. Telling users that just want to inspect an app's HTTP requests about this fairly complex tool they never heard of and probably don't need to use seems unnecessary.

- `objection` requires [a separate tool](https://github.com/NickstaDB/patch-apk) for patching split APKs while `apk-mitm` supports them out of the box.

**TL;DR:** `apk-mitm` is a lot simpler to get started with and is enough for reverse-engineering simple API-backed apps. `objection` seems like the way to go in cases where `apk-mitm` is not enough, but I believe both tools deserve to coexist. Besides, competition is a good thing. :smile:

Thanks for bringing `objection` to my attention, anyway!

**Comment by blshkv** [comment]
Cool, you can share ideas at least and not reinvent the wheel. P.s. never heard about that separate tool, it must be new. Objection did not require it in the past

**Comment by blshkv** [comment]
Btw, you ssl pinning is not the only problem with mobile apps often. There is also jail break checks. Objection allows to bypass it too. You will need frida to debug manually if automated tools failed.
